immutable/index.ts

```diff
@@ -1,8 +1,5 @@
-// @ts-ignore
-import useSWR from 'swr'
-
+import useSWR, { Middleware, SWRHook } from 'swr'
```

Comment:
only keep type imports in `immutable` and `infinite`, for reset of the generated types in `subpath/dist/src/xxx/x.d.ts` let's revisit them later